### PR TITLE
FIX  account_invoice_rounding: tax included in price + discount

### PR DESCRIPTION
**Pull request by eLBati**
See commits

### CODE REVIEW

File: account_invoice_rounding/account.py [hunk 1]
```diff
@@ -57,28 +57,11 @@ def _swedish_add_invoice_line(self, cr, uid, invoice,
...
-        tax_ids = set()
-        for line in invoice.invoice_line:
-            # invoice_line_tax_id is a many2many if you wonder about it
-            for tax in line.invoice_line_tax_id:
-                if not tax.price_include:
-                    tax_ids.add(tax.id)
-        computed_tax_ids = [tax.id for tax in invoice.tax_line]
-        return len(tax_ids) == len(computed_tax_ids)
-
     def _swedish_round_globally(self, cr, uid, invoice,
```
Comment (elicoidal) [inline account_invoice_rounding/account.py:60]: Why Swedish? Is there any special rounding that does not apply anywhere else?
If possible I would rename to _round_globally()

File: account_invoice_rounding/test/test_invoice_rounding.yml [hunk 2]
```diff
@@ -17,6 +17,37 @@
     type_tax_use: all
     tax_code_id: tax_case
 -
+  I create a Tax Code for 8%
```
Comment (elicoidal) [inline account_invoice_rounding/test/test_invoice_rounding.yml:20]: As far as I know yml tests are obsolete: @pedrobaeza can you confirm?

File: account_invoice_rounding/account.py (hunk 1 above)
Comment (eLBati) [inline account_invoice_rounding/account.py:60]: See https://en.wikipedia.org/wiki/Cash_rounding

File: account_invoice_rounding/test/test_invoice_rounding.yml (hunk 2 above)
Comment (eLBati) [inline account_invoice_rounding/test/test_invoice_rounding.yml:20]: It seems 9.0 and master use them:
https://github.com/odoo/odoo/blob/9.0/addons/stock/__openerp__.py#L79
https://github.com/odoo/odoo/blob/master/addons/stock/__openerp__.py#L94

**Comment by eLBati** [comment]
Hi @yvaucher what do you think about this?

**Review by tafaRU — APPROVED** [review]

**Comment by rafaelbn** [comment]
This PR sound quite important but alto no reviewers :thinking: 

Well we are not using v8 anymore and I cannot review it technically sorry! :cry:

**Comment by rafaelbn** [comment]
@eLBati could you rebase?

**Comment by eLBati** [comment]
It seems meanwhile yml file was removed and tests refactored to python:
https://github.com/OCA/account-invoicing/blob/8.0/account_invoice_rounding/tests/test_invoice_rounding.py

I will not refactor tests of this PR to python, but I can remove ```test_invoice_rounding.yml``` and leave the bugfix part

File: account_invoice_rounding/account.py (hunk 1 above)
Comment (yvaucher) [reply account_invoice_rounding/account.py:60]: @elicoidal that is the original name of the module and then this method as it is the "wiki-generic" name for this kind of rounding. Could be renamed `cash rounding` thought.

File: account_invoice_rounding/account.py [hunk 3]
```diff
@@ -57,28 +57,11 @@ def _swedish_add_invoice_line(self, cr, uid, invoice,
...
                                 rounded_total, delta, context=None):
         """ Add the diff to the biggest tax line
         This ajustment must be done only after all tax are computed
```
Comment (yvaucher) [inline account_invoice_rounding/account.py:63]: This docstring should probably be updated then.

**Review by yvaucher — APPROVED** [review]
This module was created on version 7.0, then IIRC in 8.0 taxes are computed differently is that right?

Thus this change would make sense.

Just don't forget to update the docstring of the method, please.

**Pull request closed without merging**